stacker/exceptions.py

```diff
@@ -128,3 +128,13 @@ def __init__(self, variable, validator, value, exception=None):
 
     def __str__(self):
         return self.message
+
+
+class UnallowedValue(Exception):
+    """Exception raised when the value provided to a variable is not allowed.
+    """
+    def __init__(self, variable, value, allowed_values, *args, **kwargs):
```

Comment:
Yea, a base exception that includes blueprint and some other data would be useful. I'm going to switch this to a `ValueError` for now and we can address this with the issue to track better exception handling